src/utils.ts

```diff
@@ -0,0 +1,413 @@
+}
+
+/**
+ * Sleep for a certain amount of time
+ *
+ * @param {number} ms The number of milliseconds to sleep for
+ * @returns {Promise<void>} A promise that resolves after the sleep
+ */
+export function sleep(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+}
+/**
+ * Converts a decimal value to its Amino string representation. Use this function when you need to create transaction messages with Dec values.
+ *
+ * @example
+ * ```
+ * decToString(0.5) => "0.500000000000000000"
+ * ```
+ *
+ * @param {number} dec The decimal value to convert.
+ * @returns {string} An Amino string representation of the decimal value.
+ */
+export function decToString(dec: number): string {
+  return dec.toFixed(18);
+}
+
+/**
+ * Convert a bech32 encoded address from one prefix to another.
+ *
+ * Note: This function works only if both chains use the same BIP-44 coin type.
+ *
+ * @param {string} address The bech32 encoded address to convert.
+ * @param {string} toPrefix The target prefix to convert the address to.
+ * @returns {string} The converted bech32 encoded address with the new prefix.
+ */
+export function convertBech32Prefix(address: string, toPrefix: string): string {
+  return bech32.encode(toPrefix, bech32.decode(address).words);
+}
+
+/**
+ * Represents an IBC response, which can be either an acknowledgement or a timeout.
+ */
+export type IbcResponse = {
+  /**
+   * The type of IBC response, either "ack" for acknowledgement or "timeout" for timeout.
+   */
+  type: "ack" | "timeout";
+
+  /**
+   * The transaction details associated with the IBC response.
+   */
+  tx: IndexedTx;
+};
+/**
+ * Retrieves the IBC responses for a given transaction response, with options to customize the timeout and check interval for the IBC response transactions. If `txResponse.code = 0` and the transaction resulted in sending IBC packets, `getIbcResponse()` will return a list of IBC acknowledgement or timeout transactions which signal whether the original IBC packet was accepted, rejected or timed-out on the receiving chain.
+ *
+ * @param {object} txResponse - The transaction response object.
+ * @param {number} [resolveResponsesTimeoutMs=120000] - The timeout in milliseconds for waiting for IBC response txs to commit on-chain. Defaults to 120000 (2 minutes).
+ * @param {number} [resolveResponsesCheckIntervalMs=15000] - The interval in milliseconds between checks when waiting for IBC response txs to commit on-chain. Defaults to 15000 (15 seconds).
+ * @returns {Promise<object>} A list of IBC acknowledgement or timeout transactions which signal whether the original IBC packet was accepted, rejected or timed-out on the receiving chain.
+ */
+export function getTxIbcResponses(
+  stargateClient: StargateClient,
+  txResponse: DeliverTxResponse,
+  resolveResponsesTimeoutMs: number = 120000,
+  resolveResponsesCheckIntervalMs: number = 15000,
+): Array<Promise<IbcResponse>> {
```

Comment:
q: Could there be multiple send_packet events on one transaction? Do we know why?

Reply:
yes! e.g. `signAndBroadcast([msgTransfer,msgTransfer])`, and also ibc contracts if we're ever going to have any. plus this worka on other chains too

Reply:
Gotcha, TIL! :D